packages/server/api/controllers/favorites.controller.js

```diff
@@ -0,0 +1,35 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+
+const getFavorites = async () => {
+  try {
+    const favorites = await knex.select().table('favorites');
+    if (favorites.length === 0) {
+      throw new HttpError('No favorites found', 404);
+    }
+    return favorites;
+  } catch (error) {
+    return error.message;
+  }
+};
+
+const createFavorites = async (body) => {
+  await knex('favorites').insert({
+    user_id: body.user_id,
+    product_id: body.product_id,
+    created_at: body.created_at,
```

Comment:
What if these values are `null`?

Reply:
Yes point I will try to make a conditions .

Reply:
I have updated this in try/catch  handaling as it shows error messages if values are null